core/src/services/seafile/backend.rs

```diff
@@ -0,0 +1,343 @@
+        }?;
+
+        let username = match &self.config.username {
+            Some(username) => Ok(username.clone()),
+            None => Err(Error::new(ErrorKind::ConfigInvalid, "username is empty")
+                .with_operation("Builder::build")
+                .with_context("service", Scheme::Seafile)),
+        }?;
+
+        let password = match &self.config.password {
+            Some(password) => Ok(password.clone()),
+            None => Err(Error::new(ErrorKind::ConfigInvalid, "password is empty")
+                .with_operation("Builder::build")
+                .with_context("service", Scheme::Seafile)),
+        }?;
+
+        let client = if let Some(client) = self.http_client.take() {
+            client
+        } else {
+            HttpClient::new().map_err(|err| {
+                err.with_operation("Builder::build")
+                    .with_context("service", Scheme::Seafile)
+            })?
+        };
+
+        Ok(SeafileBackend {
+            core: Arc::new(SeafileCore {
+                root,
+                server,
+                username,
+                password,
+                repo_name: self.config.repo_name.clone(),
+                auth_info: Arc::new(RwLock::new(AuthInfo::default())),
+                client,
+            }),
+        })
+    }
+}
+
+/// Backend for seafile services.
+#[derive(Debug, Clone)]
+pub struct SeafileBackend {
+    core: Arc<SeafileCore>,
+}
+
+#[async_trait]
+impl Accessor for SeafileBackend {
+    type Reader = IncomingAsyncBody;
+
+    type BlockingReader = ();
+
+    type Writer = SeafileWriters;
+
+    type BlockingWriter = ();
+
+    type Lister = oio::PageLister<SeafileLister>;
+
+    type BlockingLister = ();
+
+    fn info(&self) -> AccessorInfo {
+        let mut am = AccessorInfo::default();
+        am.set_scheme(Scheme::Seafile)
+            .set_root(&self.core.root)
+            .set_native_capability(Capability {
+                stat: true,
+
+                read: true,
+                read_can_next: true,
+
+                write: true,
+                write_can_empty: true,
+
+                delete: true,
+
+                list: true,
+
+                ..Default::default()
+            });
+
+        am
+    }
+
+    async fn read(&self, path: &str, _args: OpRead) -> Result<(RpRead, Self::Reader)> {
+        let resp = self.core.download_file(path).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => {
+                let size = parse_content_length(resp.headers())?;
+                Ok((RpRead::new().with_size(size), resp.into_body()))
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn stat(&self, path: &str, _args: OpStat) -> Result<RpStat> {
+        if path == "/" {
+            return Ok(RpStat::new(Metadata::new(EntryMode::DIR)));
+        }
+
+        let metadata = if path.ends_with('/') {
```

Comment:
Please don't depend on path to decide the entry mode.

Reply:
How can I determine if the file attributes and folder attributes in Seafile are two different APIs? Here's a reference to the stat function in azfile.

Reply:
Azblob is an object storage service that allows both `abc` and `abc/` to coexist simultaneously. Does Seafile offer this same capability?

Reply:
In Seafile, abc/ should be a folder, while abc is a file.

Reply:
> In Seafile, abc/ should be a folder, while abc is a file.

What will happen if we try to stat a dir with path `abc`?

Reply:
Based on the current code, it seems it will proceed with the file attribute query API, so the expected return should be "NOT_FOUND"

Reply:
Ok, this behavior also LGTM.